ludwig/utils/fs_utils.py

```diff
@@ -182,6 +182,26 @@ def safe_move_file(src, dst):
             raise
 
 
+@DeveloperAPI
+def safe_move_directory(src, dst):
+    try:
+        os.replace(src, dst)
+    except OSError as err:
+        if err.errno == errno.EXDEV:
+            # Generate a unique ID, and copy `<src>` to the target directory with a temporary name `<dst>.<ID>.tmp`.
+            # Because we're copying across a filesystem boundary, this initial copy may not be atomic.  We insert a
+            # random UUID so if different processes are copying into `<dst>`, they don't overlap in their tmp copies.
+            copy_id = uuid.uuid4()
+            tmp_dst = f"{dst}.{copy_id}.tmp"
+            shutil.copytree(src, tmp_dst)
+
+            # Atomic replace file onto the new name, and clean up original source file.
+            os.replace(tmp_dst, dst)
+            os.unlink(src)
+        else:
+            raise
```

Comment:
Is this right?

Reply:
@justinxzhao It seems to work as expected for me. Just fixed, I noticed that my commit changing this to `shutil.copytree` from `shutil.copyfile` didn't get pushed up